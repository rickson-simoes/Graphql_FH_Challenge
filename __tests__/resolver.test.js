const request = require('supertest');
const resolvers = require('../src/resolvers');

describe('Query', () => {
  it ('Deve exibir todas as contas', async () => {
    const response = await request(resolvers)
      .post('/tests')
      .send({
        query : {
          contas : {
            conta,
            saldo
          }
        }
      });

    expect(response).toBeTruthy();
  });

  it ('Deve exibir uma conta sÃ³', async () => {
    const response = await request(resolvers)
      .post('/tests')
      .send({
        query : {
          conta : 12345
        }
      });

    expect(response).toBeTruthy();
  });
});

describe('Mutations', () => {
  it ('Deve criar uma conta', async () => {
    const response = await request(resolvers)
      .post('/tests')
      .send({
        mutation : {
          criarContas : {
            conta:123,
            saldo:100
          }
        }
      });

    expect(response).toBeTruthy();
  });

  it ('Deve depositar dinheiro', async () => {
    const response = await request(resolvers)
      .post('/tests')
      .send({
        mutation : {
          depositar : {
            conta:12345,
            valor:100
          }
        }
      });

    expect(response).toBeTruthy();
  });

  it ('Deve sacar dinheiro', async () => {
    const response = await request(resolvers)
      .post('/tests')
      .send({
        mutation : {
          sacar : {
            conta:12345,
            valor:100
          }
        }
      });

    expect(response).toBeTruthy();
  });
});