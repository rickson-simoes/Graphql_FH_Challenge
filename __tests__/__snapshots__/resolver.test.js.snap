// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieves the shop items in default order 1`] = `Promise {}`;
